source/common/network/BUILD

```diff
@@ -127,10 +127,14 @@ envoy_cc_library(
 envoy_cc_library(
     name = "listener_lib",
     srcs = [
+        "base_listener_impl.cc",
         "listener_impl.cc",
+        "udp_listener_impl.cc",
```

Comment:
Now's also a good time to think about how we want to add build options.

I'm happy to have UDP default on (so our CI builds it) but make it easy for folks to opt out (especially to make things less painful for our windows build, while we work on cross-platform support.  I strongly suspect QUIC specifically but also likely UDP in general are going to work for Linux for a first pass.  WDYT?

Reply:
No real preference for me on whether we allow compile out on a first pass. Maybe as a follow up? I'm fine either way but definitely a good callout.